typed-racket-lib/typed-racket/utils/redirect-contract.rkt

```diff
@@ -0,0 +1,45 @@
+#lang racket/base
+
+(require syntax/modcollapse (for-template racket/base))
+(provide make-make-redirect-to-contract)
+
+;; This produces an identifier that expands to a local-require.  It's
```

Comment:
This sentence confused me, since nothing here returns an identifier. It seems that make-make-redirect-to-contract is a curried function that takes `varref` and `id` and returns a syntax transformer that expands to a local require.